pkg/generate/app/imagestreamlookup.go

```diff
@@ -90,17 +113,25 @@ func (r ImageStreamSearcher) Search(precise bool, terms ...string) (ComponentMat
 
 				latest := imageapi.LatestTaggedImage(stream, finalTag)
 				if latest == nil || len(latest.Image) == 0 {
+
 					glog.V(2).Infof("no image recorded for %s/%s:%s", stream.Namespace, stream.Name, finalTag)
-					componentMatches = append(componentMatches, &ComponentMatch{
-						Value:       term,
-						Argument:    fmt.Sprintf("--image-stream=%q", matchName),
-						Name:        matchName,
-						Description: fmt.Sprintf("Image stream %s in project %s", stream.Name, stream.Namespace),
-						Score:       0.5 + score,
-						ImageStream: stream,
-						ImageTag:    finalTag,
-						Meta:        meta,
-					})
+					if r.AllowMissingTags {
+						addMatch(finalTag, score, nil, false)
```

Comment:
hm.  we should only be doing this if the imagestream score was 0.0, i think?  (wondering what happens if you partially match an imagestream that has no tags, or at least doesn't have this tag)